File: cpp/7.cpp
// https://leetcode.com/problems/reverse-integer/
class Solution {
public:
    int reverse(int x) {
        int ans = 0;
        do {
            ans = ans * 10 + x % 10;
        } while (x /= 10);
        if (ans > INT_MAX || ans < INT_MIN) ans = 0;
        return ans;
    }
};

File: golang/7.go
package main

func reverse(x int) int {
	ans := 0
	for {
		ans = ans*10 + x%10
		if x /= 10; x == 0 {
			break
		}
	}
	if ans < math.MinInt32 || ans > math.MaxInt32 {
		ans = 0
	}
	return ans
}
